### PR TITLE
add example with-graphql-hasura

### PR DESCRIPTION
**Pull request by praveenweb**
This PR adds an example blog app based on the `blog-starter` theme and uses Hasura's GraphQL APIs.

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| buildDuration | 10.9s | 11s | ⚠️ +161ms |
| nodeModulesSize | 64.9 MB | 64.9 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.864 | 1.891 | ⚠️ +0.03 |
| / avg req/sec | 1340.86 | 1321.78 | -19.08 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.279 | 1.221 | -0.06 |
| /error-in-render avg req/sec | 1955.4 | 2047.9 | ⚠️ +92.5 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.49 kB | 6.49 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.58 kB | 5.58 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 275 B | 275 B | ✓ |
| Overall change | 545 B | 545 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| index.html gzip | 954 B | 954 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 947 B | 947 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| buildDuration | 11.7s | 12s | ⚠️ +300ms |
| nodeModulesSize | 64.9 MB | 64.9 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.49 kB | 6.49 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.58 kB | 5.58 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 275 B | 275 B | ✓ |
| Overall change | 545 B | 545 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 913 kB | 913 kB | ✓ |
| routerDirect.js | 905 kB | 905 kB | ✓ |
| withRouter.js | 905 kB | 905 kB | ✓ |
| Overall change | 4.48 MB | 4.48 MB | ✓ |

</details>

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| buildDuration | 13.2s | 13.2s | ⚠️ +22ms |
| nodeModulesSize | 64.9 MB | 64.9 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.261 | 2.23 | -0.03 |
| / avg req/sec | 1105.47 | 1120.9 | ⚠️ +15.43 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.493 | 1.509 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1674.58 | 1657.17 | -17.41 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.49 kB | 6.49 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.58 kB | 5.58 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 275 B | 275 B | ✓ |
| Overall change | 545 B | 545 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| index.html gzip | 954 B | 954 B | ✓ |
| link.html gzip | 961 B | 961 B | ✓ |
| withRouter.html gzip | 947 B | 947 B | ✓ |
| Overall change | 2.86 kB | 2.86 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| buildDuration | 14.1s | 14.4s | ⚠️ +291ms |
| nodeModulesSize | 64.9 MB | 64.9 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| main-HASH.js gzip | 6.49 kB | 6.49 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..0538.js gzip | 10.5 kB | 10.5 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 56.9 kB | 56.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| main-HASH.module.js gzip | 5.58 kB | 5.58 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| de003c3a9d30..dule.js gzip | 6.93 kB | 6.93 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| Overall change | 52.4 kB | 52.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 26.3 kB | 26.3 kB | ✓ |
| Overall change | 26.3 kB | 26.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| routerDirect.js gzip | 279 B | 279 B | ✓ |
| withRouter.js gzip | 278 B | 278 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js gzip | 2.05 kB | 2.05 kB | ✓ |
| hooks.js gzip | 881 B | 881 B | ✓ |
| _app.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error.js gzip | 3.25 kB | 3.25 kB | ✓ |
| Overall change | 8.22 kB | 8.22 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| index.module.js gzip | 223 B | 223 B | ✓ |
| routerDirect..dule.js gzip | 281 B | 281 B | ✓ |
| withRouter.m..dule.js gzip | 278 B | 278 B | ✓ |
| _error.module.js gzip | 2.04 kB | 2.04 kB | ✓ |
| hooks.module.js gzip | 383 B | 383 B | ✓ |
| link.module.js gzip | 1.52 kB | 1.52 kB | ✓ |
| _app.module.js gzip | 604 B | 604 B | ✓ |
| Overall change | 5.33 kB | 5.33 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 270 B | 270 B | ✓ |
| _buildManife..dule.js gzip | 275 B | 275 B | ✓ |
| Overall change | 545 B | 545 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _error.js | 875 kB | 875 kB | ✓ |
| 404.html | 4.17 kB | 4.17 kB | ✓ |
| hooks.html | 3.79 kB | 3.79 kB | ✓ |
| index.js | 875 kB | 875 kB | ✓ |
| link.js | 913 kB | 913 kB | ✓ |
| routerDirect.js | 905 kB | 905 kB | ✓ |
| withRouter.js | 905 kB | 905 kB | ✓ |
| Overall change | 4.48 MB | 4.48 MB | ✓ |

</details>

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| buildDuration | 11.2s | 11s | -164ms |
| nodeModulesSize | 66.5 MB | 66.5 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.778 | 1.908 | ⚠️ +0.13 |
| / avg req/sec | 1405.93 | 1310.57 | ⚠️ -95.36 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.061 | 1.164 | ⚠️ +0.1 |
| /error-in-render avg req/sec | 2355.81 | 2148.5 | ⚠️ -207.31 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| 677f882d2ed8..7778.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-eea7d2a..4017.js gzip | 6.72 kB | 6.72 kB | ✓ |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 87.6 kB | 87.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.08 kB | 6.08 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-b12b7b0..dule.js gzip | 5.79 kB | 5.79 kB | ✓ |
| polyfills-05..dule.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 82.6 kB | 82.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8e69b..8695.js gzip | 3.41 kB | 3.41 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-ddd176e..5566.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.66 kB | 7.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-663e6..dule.js gzip | 2.24 kB | 2.24 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-69bc264..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.29 kB | 5.29 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 275 B | 275 B | ✓ |
| _buildManife..dule.js gzip | 281 B | 281 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| index.html gzip | 944 B | 944 B | ✓ |
| link.html gzip | 954 B | 954 B | ✓ |
| withRouter.html gzip | 939 B | 939 B | ✓ |
| Overall change | 2.84 kB | 2.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| buildDuration | 12.4s | 12.8s | ⚠️ +413ms |
| nodeModulesSize | 66.5 MB | 66.5 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| 677f882d2ed8..7778.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-eea7d2a..4017.js gzip | 6.72 kB | 6.72 kB | ✓ |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 87.6 kB | 87.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.08 kB | 6.08 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-b12b7b0..dule.js gzip | 5.79 kB | 5.79 kB | ✓ |
| polyfills-05..dule.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 82.6 kB | 82.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8e69b..8695.js gzip | 3.41 kB | 3.41 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-ddd176e..5566.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.66 kB | 7.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-663e6..dule.js gzip | 2.24 kB | 2.24 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-69bc264..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.29 kB | 5.29 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 275 B | 275 B | ✓ |
| _buildManife..dule.js gzip | 281 B | 281 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | praveenweb/next.js add-hasura-example | Change |
| - | - | - | - |
| _error.js | 881 kB | 881 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.82 kB | 3.82 kB | ✓ |
| index.js | 881 kB | 881 kB | ✓ |
| link.js | 922 kB | 922 kB | ✓ |
| routerDirect.js | 916 kB | 916 kB | ✓ |
| withRouter.js | 916 kB | 916 kB | ✓ |
| Overall change | 4.52 MB | 4.52 MB | ✓ |

</details>

</details>
Commit: d1ea4cd03e239c1107acf6d0ee848667be44d9cb

File: examples/with-graphql-hasura/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.env
+.now
```
Comment (leerob) [inline examples/with-graphql-hasura/.gitignore:2]:
```suggestion
.vercel
```

File: examples/with-graphql-hasura/README.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+You should now be able to see the draft post. To exit the preview mode, you can click **Click here to exit preview mode** at the top.
+
+### Step 7. Deploy Hasura
+
+To deploy to production, you must first deploy Hasura on a public URL. The Hasura backend for our demo at https://next-blog-hasura.now.sh/ is deployed to Heroku ([here’s the documentation](https://hasura.io/docs/1.0/graphql/manual/getting-started/heroku-simple.html)).
+
+After setting up Hasura on Heroku, copy the Heroku app URL to apply the migrations
+
+```bash
+hasura migrate apply --endpoint https://hasura-blog-backend.herokuapp.com
```
Comment (leerob) [inline examples/with-graphql-hasura/README.md:154]: Great details here! ❤️

File: examples/with-graphql-hasura/hasura/metadata/version.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+version: 2
```
Comment (leerob) [inline examples/with-graphql-hasura/hasura/metadata/version.yaml:1]: Might also be worth mentioning this is self-hosted Hasura.

File: examples/with-graphql-hasura/README.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+```
+
+```bash
+hasura metadata apply --endpoint https://hasura-blog-backend.herokuapp.com
```
Comment (leerob) [inline examples/with-graphql-hasura/README.md:158]: Should this use Hasura Cloud now?

**Review by leerob — CHANGES_REQUESTED** [review]
In my opinion, it feels like there's a lot going on here. We have basically the same `blog-starter` template that is used for other CMS providers, but we're trying to adapt it for Hasura (which I would classify as more than a CMS). 

I'd love to see this paired down to just `with-hasura` that focused specifically on the integration with Hasura. If they want to turn it into a blog, then they can use the existing `blog-starter` examples.

Open to other opinions!

**Comment by lfades** [comment]
Closing as stale, feel free to update the PR with the requested changes and I'll open it again!

**Pull request closed without merging**